llvm/lib/IR/Verifier.cpp

```diff
@@ -2160,6 +2160,64 @@ void Verifier::verifyFunctionAttrs(FunctionType *FT, AttributeList Attrs,
            V);
   }
 
+  if (Attrs.hasFnAttr("aarch64_sme_zt0_new")) {
```

Comment:
Rather than having to spell out all possible combinations here, my preference would be to do the same thing as you did for the assert, where the error is more generic, but where the check itself is simpler by using the exclusive-or operator (and therefore less error prone to missing a case).